File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int beams = 0, prevs = 0, lasers;
        for (string row : bank){
            lasers = 0;
            for (char c : row){
                if (c == '1') lasers++;
            }
            if (lasers > 0){
                beams += prevs*lasers;
                prevs = lasers;
            }
        }
        return beams;
    }
};

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        prevLasers = 0
        beams = 0
        for row in bank:
            lasers = row.count('1')
            if lasers:
                beams += prevLasers*lasers
                prevLasers = lasers
        return beams